Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -635,13 +619,49 @@ void CGPathAddRelativeArc(
     UNIMPLEMENTED();
 }
 
+static inline D2D_RECT_F __CGRectToD2D_F(CGRect rect) {
+    return {
+        rect.origin.x, rect.origin.y, rect.origin.x + rect.size.width, rect.origin.y + rect.size.height,
+    };
+}
+
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 void CGPathAddRoundedRect(
     CGMutablePathRef path, const CGAffineTransform* transform, CGRect rect, CGFloat cornerWidth, CGFloat cornerHeight) {
-    UNIMPLEMENTED();
+    RETURN_IF(!path);
+
+    // First we must transform the coordinates of the rectangle if necessary.
+    if (transform) {
+        rect = CGRectApplyAffineTransform(rect, *transform);
+    }
+
+    D2D1_RECT_F rectangle = __CGRectToD2D_F(rect);
+    D2D1_ROUNDED_RECT roundedRectangle = { rectangle, cornerWidth, cornerHeight };
+
+    ComPtr<ID2D1Factory> factory;
+    FAIL_FAST_IF_FAILED(_CGGetD2DFactory(&factory));
+    ComPtr<ID2D1RoundedRectangleGeometry> rectangleGeometry;
+
+    FAIL_FAST_IF_FAILED(factory->CreateRoundedRectangleGeometry(&roundedRectangle, &rectangleGeometry));
+
+    path->ClosePath();
+
+    // Create temp vars for new path/sink
+    ComPtr<ID2D1PathGeometry> newPath;
+    ComPtr<ID2D1GeometrySink> newSink;
+
+    // Open a new path that the contents of the old path will be streamed into. We cannot re-use the same path as it is now closed
+    // and cannot be opened again.
+    FAIL_FAST_IF_FAILED(factory->CreatePathGeometry(&newPath));
+    FAIL_FAST_IF_FAILED(newPath->Open(&newSink));
+
+    FAIL_FAST_IF_FAILED(rectangleGeometry->CombineWithGeometry(path->GetPathGeometry().Get(), D2D1_COMBINE_MODE_XOR, NULL, newSink.Get()));
+
```

Comment:
Here's a thought: What if we "simplify" an ellipse or rounded rect geometry into the sink for our existing geometry?

That also allows us to propagate transform via D2DTransformedGeometry.